### ISSUE TITLE
Allow adding only certain items from a recipe to a shopping list

### ISSUE THREAD

**Issue by edcous** [vabene1111/recipes#305]
**Is your feature request related to a problem? Please describe.**
When creating a meal plan, it adds every item to the shopping list.
**Describe the solution you'd like**
This becomes an issue when it adds things like water to the shopping list when you already have it, it would be amazing to have a popup to remove items from the shopping list before creating it, or just select the items you want to add.
**Describe alternatives you've considered**
Just checking off the items you don't want works, but makes the shopping list a bit cluttered and not super intuitive.
**Additional context**
<img width="934" alt="image" src="https://user-images.githubusercontent.com/49954272/103467984-46f6cd00-4d1a-11eb-8843-c39819b3d84e.png">
example of something that i don't need on the shopping list ^

**Comment by vabene1111** [comment]
i am actually dealing with this every time i create a shopping list myself 😂 i usually just check the items since i export the list to google keep afterwards anyway and checked items are ignored in the export. 

This is definitly something i want to fix, probably by using a property on the food itself where you can set it to not show on shopping lists. This will likely be implemented together with some other features related to food data

**Comment by edcous** [comment]
Ah ok, excited to see this implemented :)

**Comment by philip-n** [comment]
Let me add my voice to the people interested in this :)

I just started importing recipes into Tandoor (love it so far!) and one thing I noticed was that most recipes use generic stuff like "oil", "pepper and salt", various spices, ... that you usually do not buy on a per-meal-basis.

Instead of adding some flag of "which items to not add to the shopping list" on a per recipe bases, I'd vote in favor of adding such a flag to each ingredient directly (as I now see @vabene1111 already mentioned). This would also make migrations of existing recipes easier, I guess.

Only caveat: It would be nice if one could still manually add the according ingredient to the shopping list (i.e. the flag should only prevent automatic adding).

Other opinions?

**Comment by vabene1111** [comment]
That is actually exactly the way it is implemented currently. you can flag each food (ingredients are alywas a combination of unit, amount and food) to be ignored at shopping list creation. 

I guess i just forgot about this issue when i implemented the new shopping list. Also it is planned to have a "quick edit" menu to quickly set this setting directly from the shopping list.

Let me know if anything needs changing or if you need further help figuring this out

**Comment by philip-n** [comment]
Ah, that's great - thanks for pointing it out! I simply hadn't clicked on an ID in the food view yet, as I did not think there would be something interesting to find. Or more precise: Didn't think about "what will a clickable ID lead to" it at all.

**Small proposal**: In the "Food" view, make it more visible that the food can be edited (e.g. by making the name of the ingredient itself clickable, or by displaying a clickable pen icon next to the food).

**Bigger proposal**: In the "Food" view, show the data associated with each food (supermarket category, icon for "ignore shopping", etc).

And while I'm at it: What is the meaning of the "recipe" dropdown when editing a single food?

*Let me know if you prefer to keep issues clean and not add follow-up questions like this.*

**Comment by vabene1111** [comment]
firstly follow up questions are totally fine, if actual work needs to be done i will usually just create a new issue with just the work and reference the discussion

secondly: @smilerz is currently working on redesigning the keyword view. once that is done the same technique will be used across all views (mainly units and foods) and more info will be shown.

third(ly?!?): recipe allows to link a recipe to an ingredient. I use it for pizza dough for example. I have the dough as one recipe and I add it to all pizza recipes. It is planned to add real component based recipes but that is not yet done. A recipe linked this way will be displayed in the ingredient list with a link. (i guess this needs a help text, like so many things 😂 )

**Comment by vabene1111** [comment]
PS: will not open a new issue as reworking the food page is planned anyway and i can remember this (at least i hope)

**Comment by smilerz** [comment]
RE: marking items not for shopping list - there are a couple ways to solve this and probably should support each method:
- adding individual ingredients to the list
- batch adding ingredients to the list (multi-select)
- permanently marking an ingredient as ignored
- creating a 'pantry' that allows you to manage ingredients on hand  (I'm hoping to do this as it would allow a 'what can i make' view)

**Comment by vabene1111** [comment]
yes a pantry/stock management feature would be cool. Would nicely integrate with grocy as well.

the other features are also nice but for me at least not a very high priority for now
